js/source/worker_tile.js

```diff
@@ -260,20 +258,8 @@ WorkerTile.prototype.redoPlacement = function(angle, pitch, showCollisionBoxes)
     };
 };
 
-function isBucketEmpty(bucket) {
-    for (var programName in bucket.arrayGroups) {
-        var programArrayGroups = bucket.arrayGroups[programName];
-        for (var k = 0; k < programArrayGroups.length; k++) {
-            var programArrayGroup = programArrayGroups[k];
-            for (var layoutOrPaint in programArrayGroup) {
-                var arrays = programArrayGroup[layoutOrPaint];
-                for (var bufferName in arrays) {
-                    if (arrays[bufferName].length > 0) return true;
-                }
-            }
-        }
-    }
-    return false;
+function isBucketNonEmpty(bucket) {
```

Comment:
Keeping function names in the affirmative (`isBucketEmpty`) will keep our codebase more readable.